src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/PeriodicBatchingSink.cs

```diff
@@ -187,24 +130,31 @@ async Task LoopAsync()
                 else
                 {
                     isEagerBatch = false;
+
                     await _targetSink.EmitBatchAsync(_currentBatch).ConfigureAwait(false);
-                }
 
-                _currentBatch.Clear();
-                _batchScheduler.MarkSuccess();
+                    _currentBatch.Clear();
+                    _batchScheduler.MarkSuccess();
+                }
             }
             catch (Exception ex)
             {
                 SelfLog.WriteLine($"PeriodicBatchingSink ({_targetSink}) failed emitting a batch: {ex}");
```

Comment:
could all the selflog emissions be managed by some helper class? Maybe as virtuals in a base class ?

Reply:
Factored out but kept it pretty simple, I think eventually we'll be able to deal with more interesting use cases via the queue monitoring and (hypothetical) fallback mechanisms.

Will hit the button, CI is green :-) 🎉